pennylane_qrack/qrack_device.py

```diff
@@ -281,23 +387,45 @@ def _apply_gate(self, op):
             self._state.mtrx([1, 0, 0, cmath.exp(1j * -par[0])], device_wires.labels[0])
-        elif opname == "ControlledPhaseShift":
+        elif opname == "C(PhaseShift)" or opname == "C(U1)":
+            self._state.mtrx(device_wires.labels[:-1], [1, 0, 0, cmath.exp(1j * par[0])], device_wires.labels[-1])
+        elif opname == "C(PhaseShift).inv" or opname == "C(U1).inv":
+            self._state.mtrx(device_wires.labels[:-1], [1, 0, 0, cmath.exp(1j * -par[0])], device_wires.labels[-1])
+        elif opname == "ControlledPhaseShift" or opname == "C(ControlledPhaseShift)" or opname == "CPhase" or opname == "C(CPhase)":
             self._state.mcmtrx(device_wires.labels[:-1], [1, 0, 0, cmath.exp(1j * par[0])], device_wires.labels[-1])
-        elif opname == "ControlledPhaseShift.inv":
+        elif opname == "ControlledPhaseShift.inv" or opname == "C(ControlledPhaseShift).inv" or opname == "CPhase.inv" or opname == "C(CPhase).inv":
             self._state.mcmtrx(device_wires.labels[:-1], [1, 0, 0, cmath.exp(1j * -par[0])], device_wires.labels[-1])
         elif opname == "U2":
-            self._state.mtrx([1, cmath.exp(1j * par[1]), cmath.exp(1j * par[0]), cmath.exp(1j * (par[0] + par[1]))], device_wires.labels[-1])
+            self._state.mtrx([1, cmath.exp(1j * par[1]), cmath.exp(1j * par[0]), cmath.exp(1j * (par[0] + par[1]))], device_wires.labels[0])
         elif opname == "U2.inv":
-            self._state.mtrx([1, cmath.exp(1j * -par[1]), cmath.exp(1j * -par[0]), cmath.exp(1j * (-par[0] - par[1]))], device_wires.labels[-1])
+            self._state.mtrx([1, cmath.exp(1j * -par[1]), cmath.exp(1j * -par[0]), cmath.exp(1j * (-par[0] - par[1]))], device_wires.labels[0])
+        elif opname == "C(U2)":
+            self._state.mcmtrx(device_wires.labels[:-1], [1, cmath.exp(1j * par[1]), cmath.exp(1j * par[0]), cmath.exp(1j * (par[0] + par[1]))], device_wires.labels[-1])
+        elif opname == "C(U2).inv":
+            self._state.mcmtrx(device_wires.labels[:-1], [1, cmath.exp(1j * -par[1]), cmath.exp(1j * -par[0]), cmath.exp(1j * (-par[0] - par[1]))], device_wires.labels[-1])
         elif opname == "U3":
             self._state.u(device_wires.labels[-1], par[0], par[1], par[2])
         elif opname == "U3.inv":
             self._state.u(device_wires.labels[-1], -par[0], -par[1], -par[2])
-        else:
+        elif opname == "C(U3)":
+            self._state.mcu(device_wires.labels[:-1], device_wires.labels[-1], par[0], par[1], par[2])
+        elif opname == "C(U3).inv":
+            self._state.mcu(device_wires.labels[:-1], device_wires.labels[-1], -par[0], -par[1], -par[2])
+        elif opname == "QFT":
+            self._state.qft(device_wires.labels)
+        elif opname == "QFT.inv":
+            self._state.iqft(device_wires.labels)
+        elif opname != "Identity" and opname != "Identity.inv" and opname != "C(Identity)" and opname != "C(Identity).inv":
             raise DeviceError(
```

Comment:
Up to you, but you can use the more concise "f-string" convention for this: 

```py
raise DeviceError(f"Operation {opname} is not supported on a {self.short_name} device.")
```

Reply:
Done. (Thanks!)